cmd/promtool/backfill.go

```diff
@@ -0,0 +1,193 @@
+			return et, err
+		}
+	}
+
+	if err := p.s.Err(); err != nil {
+		return 0, err
+	}
+	return 0, io.EOF
+}
+
+func getMinAndMaxTimestamps(p textparse.Parser) (int64, int64, error) {
+	var maxt, mint int64 = math.MinInt64, math.MaxInt64
+
+	for {
+		entry, err := p.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return 0, 0, errors.Wrap(err, "next")
+		}
+
+		if entry != textparse.EntrySeries {
+			continue
+		}
+
+		_, ts, _ := p.Series()
+		if ts == nil {
+			return 0, 0, errors.Errorf("expected timestamp for series got none")
+		}
+
+		if *ts > maxt {
+			maxt = *ts
+		}
+		if *ts < mint {
+			mint = *ts
+		}
+	}
+
+	if maxt == math.MinInt64 {
+		maxt = 0
+	}
+	if mint == math.MaxInt64 {
+		mint = 0
+	}
+
+	return maxt, mint, nil
+}
+
+func createBlocks(input *os.File, mint, maxt, maxSamplesInAppender int64, outputDir string) error {
+
+	blockDuration := tsdb.DefaultBlockDuration
+	mint = blockDuration * (mint / blockDuration)
+
+	for t := mint; t <= maxt; t = t + blockDuration {
+
+		err := func() error {
+			// TODO(aSquare14) : Print block metadata.
+			w, err := tsdb.NewBlockWriter(log.NewNopLogger(), outputDir, blockDuration)
+			if err != nil {
+				return errors.Wrap(err, "block writer")
+			}
+			defer func() {
+				err = tsdb_errors.NewMulti(err, w.Close()).Err()
+			}()
+
+			ctx := context.Background()
+			app := w.Appender(ctx)
+
+			if _, errReset := input.Seek(0, 0); errReset != nil {
+				return errors.Wrap(errReset, "seek file")
+			}
+			p := NewOpenMetricsParser(input)
+
+			tsUpper := t + blockDuration
+			var samplesCount int64
+
+			for {
+				e, errP := p.Next()
+				if errP == io.EOF {
+					break
+				}
+				if errP != nil {
+					return errors.Wrap(errP, "parse")
+				}
+				if e != textparse.EntrySeries {
+					continue
+				}
+
+				l := labels.Labels{}
+				p.Metric(&l)
+				_, ts, v := p.Series()
+				if ts == nil {
+					return errors.Errorf("expected timestamp for series %v, got none", l)
+				}
+				if *ts < t || *ts >= tsUpper {
+					continue
+				}
+
+				if _, err := app.Add(l, *ts, v); err != nil {
+					return errors.Wrap(err, "add sample")
+				}
+
+				samplesCount++
+				if samplesCount < maxSamplesInAppender {
+					continue
+				}
+
+				if err := app.Commit(); err != nil {
+					return errors.Wrap(err, "commit")
+				}
+
+				app = w.Appender(ctx)
+				samplesCount = 0
+			}
+
+			if err := app.Commit(); err != nil {
+				return errors.Wrap(err, "commit")
+			}
+
+			if _, errF := w.Flush(ctx); errF != nil && errF != tsdb.ErrNoSeriesAppended {
+				return errors.Wrap(errF, "flush")
+			}
+			return nil
+		}()
+
+		if err != nil {
+			return errors.Wrap(err, "process blocks")
+		}
+	}
+	return nil
+}
+
+func backfill(maxSamplesInAppender int64, input *os.File, outputDir string) (err error) {
+
+	p := NewOpenMetricsParser(input)
+	maxt, mint, errTs := getMinAndMaxTimestamps(p)
+	if errTs != nil {
+		return errors.Wrap(errTs, "error getting min and max timestamp")
```

Comment:
```suggestion
		return errors.Wrap(errTs, "getting min and max timestamp")
```
We know it's error since we are wrapping it :hugs: